code/game/machinery/portable_turret.dm

```diff
@@ -511,61 +379,83 @@ Status: []<BR>"},
 		popDown()
 		return
 
+	last_fired = 0
+	if(attacked)
+		attacked--
+
 	var/list/targets = list()		   // list of primary targets
 	var/list/secondarytargets = list() // targets that are least important
 
-	if(src.check_anomalies) // if its set to check for xenos/carps, check for non-mob "crittersssss"(And simple_animals)
-		for(var/mob/living/simple_animal/C in view(7,src))
-			if(C.stat)
+
+
+	if(emagged)
+		for(var/mob/living/L in view(12, src))
+			if(L.flags & INVULNERABLE)
 				continue
-			if(C.flags & INVULNERABLE)
+			if(istype(L, /mob/living/silicon/robot/mommi) || istype(L, /mob/living/silicon/pai)) //mommis are always safe, can't shoot pAIs
 				continue
-			// Ignore lazarus-injected mobs.
-			if(dd_hasprefix(C.faction, "lazarus"))
+			if(L.isDead())         //if the turret is emagged, skip all the fancy target picking stuff
 				continue
-			targets += C
+			if(L.isUnconscious())  //and focus on murdering everything
+				secondarytargets += L
+			else
+				targets += L
 
-	for (var/mob/living/carbon/C in view(7,src)) // loops through all living carbon-based lifeforms in view(12)
-		if(C.flags & INVULNERABLE)
-			continue
-		if(istype(C, /mob/living/carbon/alien) && src.check_anomalies) // git those fukken xenos
-			if(!C.stat) // if it's dead/dying, there's no need to keep shooting at it.
+	else
+		if(src.check_anomalies) // if its set to check for xenos/carps, check for non-mob "crittersssss"(And simple_animals)
+			for(var/mob/living/simple_animal/C in view(7,src))
+				if(C.stat)
+					continue
+				if(C.flags & INVULNERABLE)
+					continue
+				// Ignore lazarus-injected mobs.
+				if(dd_hasprefix(C.faction, "lazarus"))
+					continue
 				targets += C
 
-		else
-			if(emagged) // if emagged, HOLY SHIT EVERYONE IS DANGEROUS beep boop beep
-				targets += C
+		for (var/mob/living/carbon/C in view(7,src)) // loops through all living carbon-based lifeforms in view(7)
+			if(C.flags & INVULNERABLE)
+				continue
+
 			else
-				if (C.stat || C.handcuffed) // if the perp is handcuffed or dead/dying, no need to bother really
-					continue // move onto next potential victim!
+				if(istype(C, /mob/living/carbon/alien) && src.check_anomalies) // git those fukken xenos
+					if(!C.stat) // if it's dead/dying, there's no need to keep shooting at it. //if there are still standing xenos
+						targets += C
+					else if(C.stat == UNCONSCIOUS) // finish purging if nothing else to do
+						secondarytargets += C
 
-				var/dst = get_dist(src, C) // if it's too far away, why bother?
-				if (dst > 7)
-					continue
+				else
+					if (C.stat || C.handcuffed) // if the perp is handcuffed or dead/dying, no need to bother really
```

Comment:
Use `restrained()` for the handcuffed check, `isUnconscious()` for the stat check.